felt/src/lib.rs

```diff
@@ -162,6 +162,16 @@ mod test {
             let p = &BigUint::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
             prop_assert!(&x.to_biguint() < p);
         }
+
+        #[test]
+        // Property-based test that ensures, for 100 felt values that are randomly generated each time tests are run, that a felt created using Felt::from_bytes_be doesn't fall outside the range [0, p].
+        // In this and some of the following tests, The value of {x} can be either [0] or a very large number, in order to try to overflow the value of {p} and thus ensure the modular arithmetic is working correctly.
+        fn from_bytes_be_in_range(ref x in "(0|[1-9][0-9]*)") {
+            let x = &Felt::from_bytes_be(x.as_bytes());
+            let p = &Felt::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
```

Comment:
The reason why I dint use `BigUint` is because in order to do that I would have to  convert the x value to BigUint using `x.to_biguint()`. It is not the current case, but felt methods may change in the future, and `to_biguint()` could change the inner value of x , for example, applying modulo to it before the conversion, in this case, it would defeat the purpose of the property test, as we want to make sure that `Felt::from_bytes_be` returns a number that is within the field, and the test would pass even if this isnt true.

Reply:
But we're still comparing with a `Felt` that is `0`. If the comparison below is true, then we have a bug. The closest you can get is comparing to `Felt::max()` with `<=`.
Regarding `to_biguint` changing, no, it will never apply modulo at that point because it's supposed to always be in that range between operations. That's pretty much the point of the abstraction. `to_biguint` must never modify the contained value.